edward/inferences.py

```diff
@@ -180,16 +178,24 @@ def initialize(self, n_iter=1000, n_data=None, n_print=100,
         self.n_print = n_print
         self.loss = tf.constant(0.0)
 
-        # Set shape of data placeholders according to batch size.
-        if not isinstance(self.model, StanModel):
-            for value in six.itervalues(self.data):
-                value.set_shape([self.n_data] + get_dims(value)[1:])
+        if n_data is not None and not isinstance(self.model, StanModel):
+            # Re-assign data to batch tensors, with size given by `n_data`.
+            values = list(six.itervalues(self.data))
+            slices = tf.train.slice_input_producer(values)
+            batches = tf.train.batch(slices, self.n_data)
+            if not isinstance(batches, list):
+                # `tf.train.batch` returns tf.Tensor if `slices` is a
+                # list of size 1.
+                batches = [batches]
+
+            self.data = {key: value for key, value in
+                         zip(six.iterkeys(self.data), batches)}
 
         loss = self.build_loss()
         if optimizer is None:
             var_list = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES,
                                          scope=scope)
-            # Use ADAM with a decaying scale factor
+            # Use ADAM with a decaying scale factor.
             global_step = tf.Variable(0, trainable=False)
             starter_learning_rate = 0.1
             learning_rate = tf.train.exponential_decay(starter_learning_rate,
```

Comment:
I don't think so. We've seen it help in practice for convergence on toy problems in the `examples/` directory. We can change it in another pull request during heavier investigation.